src/fighter.rs

```diff
@@ -17,6 +17,14 @@ use crate::{
     player::Player,
 };
 
+pub struct FighterPlugin;
+
+impl Plugin for FighterPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_system_to_stage(CoreStage::PostUpdate, attachment_system);
```

Comment:
Noting that you might want to put this in `CoreStage::Update`, but I could be wrong, will return for a closer look.

Reply:
All the state handler functions run in `Update`, so I think `PostUpdate` is right.

Reply:
This actually solves a problem when grabbing a item with the player facing left, the attached item appears facing right for one frame and then facing left, this is because the facing value for the item is first set to default/right, and just when the system is run it fixes it.